web/src/beta/features/Editor/Settings/hooks.ts

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect, useMemo } from "react";
+
+import { NLSLayer } from "@reearth/services/api/layersApi/utils";
+import { Page } from "@reearth/services/api/storytellingApi/utils";
+
+type SettingProps = {
+  onPageUpdate?: (id: string, layers: string[]) => void;
+  layers?: NLSLayer[];
+  selectedPage?: Page;
+};
+export default ({ layers, selectedPage, onPageUpdate }: SettingProps) => {
+  const pageId = selectedPage?.id;
+
+  const allLayerIds = useMemo(() => {
+    return layers?.map(layer => layer.id) || [];
+  }, [layers]);
+
+  const selectedLayerIds = useMemo(() => {
+    return selectedPage?.layersIds || [];
+  }, [selectedPage]);
+
+  const allLayersSelected = useMemo(() => {
+    return selectedLayerIds.length === allLayerIds.length;
+  }, [selectedLayerIds, allLayerIds]);
```

Comment:
This isn't guaranteed to be correct. If we have LayerA, LayerB and LayerC, then select all , then delete LayerA from the project, it'll think we haven't selected all.
I feel like maybe there is a better solution somewhere, but for now an easy fix is just to make sure `selectedLayerIds.length >= allLayerIds.length`